presto-hive/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -176,6 +177,50 @@ private static Properties getHiveSchema(
         return schema;
     }
 
+    public static Properties getHiveSchema(Map<String, String> serdeParameters, Map<String, String> tableParameters)
+    {
+        Properties schema = new Properties();
+        for (Map.Entry<String, String> param : serdeParameters.entrySet()) {
+            schema.setProperty(param.getKey(), (param.getValue() != null) ? param.getValue() : "");
+        }
+        for (Map.Entry<String, String> entry : tableParameters.entrySet()) {
+            // add non-null parameters to the schema
+            if (entry.getValue() != null) {
+                schema.setProperty(entry.getKey(), entry.getValue());
+            }
+        }
+        return schema;
+    }
+
+    /**
+     * Recreates partition schema based on the table schema and the column
+     * coercions map.
+     *
+     * partitionColumnCount is needed to handle cases when the partition
+     * has less columns than the table.
+     *
+     * If the partition has more columns than the table does, the partitionColumns
+     * map is expected to contain information for the missing columns.
+     */
+    public static List<Column> reconstructPartitionSchema(List<Column> tableSchema, int partitionColumnCount, Map<Integer, Column> partitionColumns)
```

Comment:
What about calling the third parameter `partitionSchmeaOverride`? -- basically it defines "column overrides" at partition level right?

Reply:
Let me call it `partitionSchemaDifference`, to be consistent with the field name in the `HiveSplit`